jkq/ddsim/circuitsimulator.py

```diff
@@ -0,0 +1,113 @@
+"""Backend for DDSIM."""
+
+import logging
+import time
+import uuid
+
+from qiskit.providers import BackendV1, Options
+from qiskit import qobj, QiskitError
+from qiskit.providers.models import BackendConfiguration, BackendStatus
+from qiskit.result import Result
+
+from .jkqjob import JKQJob
+from .jkqerror import JKQSimulatorError
+
+from jkq.ddsim import pyddsim
+
+logger = logging.getLogger(__name__)
+
+
+class CircuitSimulator(BackendV1):
+    """Python interface to JKQ DDSIM"""
+
+    @classmethod
+    def _default_options(cls) -> Options:
+        return Options(
+            shots=None,
+            return_statevector=True
+        )
+
+    def __init__(self, configuration=None, provider=None):
+        conf = {
+            'backend_name': 'qasm_simulator',
+            'backend_version': '1.4.3',
+            'url': 'https://github.com/iic-jku/ddsim',
+            'simulator': True,
+            'local': True,
+            'description': 'JKQ DDSIM C++ simulator',
+            'basis_gates': ['id', 'u0', 'u1', 'u2', 'u3', 'cx', 'x', 'y', 'z', 'h', 's', 't', 'snapshot'],
+            'memory': False,
+            'n_qubits': 64,
+            'coupling_map': None,
+            'conditional': False,
```

Comment:
We do support conditional operations, don't we? Enabling this might require to set some other settings too. See https://qiskit.org/documentation/stubs/qiskit.providers.models.BackendConfiguration.html.